client/src/components/Header/Header.test.js

```diff
@@ -184,34 +184,37 @@ describe('<AuthOrProfile />', () => {
   });
 });
 
-const navigationLinks = (component, navItem) => {
-  return component.props.children[navItem].props;
+const navigationLinks = (component, key) => {
+  const target = component.props.children.find(
+    child => child && child.key === key
+  );
+  return target.props;
```

Comment:
Hey Nick, quick 'philosophical' note. We should probably move to a higher-level, user focused approach to testing React components. Rather than looking at the details of the props, we want to be asking "What is the order of the listitems here?"  So, we'd want to do stuff like this: https://stackoverflow.com/questions/57435680/whats-the-idiomatic-way-of-testing-a-list-with-dynamic-content-using-react-test

Reason being, it forces us to think about accessibility and UX, rather than implementation details.  

Having said all that, this PR was a definite improvement over what we had.